RFC-lifecycle.md

```diff
@@ -0,0 +1,186 @@
+   to specify exceptions, i.e. hosts which should still be reachable.
+3. Add `nock.forbidUnmockedRequests()` which causes any unmocked request
+   to be considered a programmer error.
+4. When Nock blocks requests to mocked hostnames, throw the error instead.
+5. Emit a deprecation warning from `nock.disableNetConnect()`.
+6. Emit a deprecation warning when `nock.enableNetConnect()` is invoked with an
+   argument.
+
+#### Discussion
+
+`nock.simulateUnreachability()` emits an unreachable-like error through normal
+HTTP error channels.
+
+In contrast, `nock.forbidUnmockedRequests()` throws an assertion error – as if
+it's a type error. The error is not emitted through the HTTP client. That way
+it bubbles up to the test runner instead of being handled by the client
+library and application.
+
+`nock.simulateUnreachability({ allowedHosts: ['localhost'] })` replaces
+`nock.enableNetConnect('localhost')`, while clarifying that the call causes
+some requests to be blocked.
+
+Okay, so I'm not thrilled with these proposals. While these use cases are
+important and need to be distinguished, having similar behavior triggered by
+two different functions is confusing. Let me try to propose an alternative
+with a combined API.
+
+### Use case 4: Handling unmocked requests (proposal 2)
+
+#### Use cases
+
+- Simulate network connection failure
+- Forbid unmocked requests (Closes #844)
+
+#### Proposed solution
+
+1. Add a new function that controls nock's behavior on unmocked requests:
+   `nock.whenUnmocked({ callThrough, simulateUnreachable, fail })`.
+2. Each of `callThrough`, `simulateUnreachable`, and `fail` is an optional
+   array. The array can be a mix of literal strings, the wildcard string, and
+   regexes.
+3. Calling `nock.whenUnmocked()` with no parameters is equivalent to
+   `nock.whenUnmocked({ callThrough: '*' })`. This is also nock's initial state.
+4. The order of precedence is: `callThrough`, `simulateUnreachable`, `fail`.
+   1. An unmocked request hitting a host matching `callThrough` is let through.
+   2. An unmocked request hitting a host matching `simulateUnreachable` emits
+      an unreachable-like error through normal HTTP error channels.
+   3. An unmocked request hitting a host matching `fail` is considered a
+      programmer error. Either the test or the application has done something
+      wrong. The error is thrown – as if a TypeError – so it bubbles up to
+      the test runner.
+
+#### Discussion
+
+As `simulateUnreachable` generates an error which can be handled by the client
+library and application code, this option is suitable for _testing how an
+application handles unreachable hosts_.
+
+By putting a wildcard match in the `fail` list, you know your tests will
+immediately choke if they try to hit live servers. Since the client library
+and application are kept out of the loop, this _provides confidence that a
+mock is configured for every request made by the application and tests_.
```

Comment:
`whenUnmocked` feels like it should be an event handler somehow ... 🤔 I’m not sure about this one, I feel there is a better way. Sorry I know that’s not helpful but all I got right now 🤷‍♂️